indexer/scripts/elastic-conf.py

```diff
@@ -96,6 +116,55 @@ def main_loop(self) -> None:
             logger.error("One or more configurations failed. Check logs for details.")
             return
 
+    def repository_exists(self, es: Elasticsearch) -> bool:
+        repo_name = self.es_snapshot_repo
+        try:
+            response = es.snapshot.get_repository(name=repo_name)
+            return repo_name in response
+        except Exception:
+            logger.exception("Error validating repository: %s", repo_name)
+        return False
+
+    def register_repository(self, es: Elasticsearch) -> bool:
+        status = False
+        repo_type = self.es_snapshot_repo_type
+        if not repo_type:
+            logger.error("Repository type must be either 'fs' or 's3'")
+            return status
+
+        if self.repository_exists(es):
+            logger.info("%s repository already exists.", repo_type)
+            return True
+
+        if repo_type == "s3":
+            settings = {
+                "bucket": self.es_snapshot_s3_bucket,
+                "endpoint": self.es_snapshot_s3_endpoint,
+            }
+        elif repo_type == "fs":
+            settings = {"location": self.es_snapshot_fs_location}
+        else:
+            logger.error("Unsupported repository type: %s", repo_type)
+            return status
+
+        try:
+            response = es.snapshot.create_repository(
+                name=self.es_snapshot_repo,
+                type=repo_type,
+                settings=settings,
+            )
+
+            if response and response.get("acknowledged", False):
+                logger.info("%s repository registered successfully.", repo_type)
+                status = True
+            else:
+                logger.error("Failed to register %s repository.", repo_type)
+        except Exception:
+            logger.exception(
+                "Exception occurred while registering %s repository", repo_type
+            )
+        return status
```

Comment:
```suggestion
            logger.exception(
                "Failed to register repository: %s", self.es_snapshot_repo
            )
        return False
```

Of course, we can also include `self.es_snapshot_repo_type` in the error message (e.g. `Failed to register repository: fs://mc_story_backups`) but it's important to include name.